pkg/virtctl/vnc/vnc.go

```diff
@@ -97,22 +106,35 @@ func (o *VNC) Run(cmd *cobra.Command, args []string) error {
 
 	// execute remote viewer
 	go func() {
-		cmnd := exec.Command("remote-viewer", fmt.Sprintf("vnc://127.0.0.1:%d", port))
-		err := cmnd.Run()
+		args := []string{fmt.Sprintf("vnc://127.0.0.1:%d", port)}
+		if glog.V(4) {
+			args = append(args, "--debug")
+			glog.Infof("remote-viewer commandline: %v", args)
+		}
+
+		cmnd := exec.Command("remote-viewer", args...)
+		output, err := cmnd.CombinedOutput()
 		if err != nil {
 			glog.Errorf("remote-viewer execution encountered an error: %v", err)
 		}
+		if err != nil || glog.V(2) {
+			glog.Errorf("remote-viewer: %v", string(output))
+		}
 		viewResChan <- err
 	}()
 
+	glog.Infof("connection timeout: %v", LISTEN_TIMEOUT)
+	// exit early if spawning remote-viewer fails
+	ln.SetDeadline(time.Now().Add(LISTEN_TIMEOUT))
```

Comment:
You can keep the timeout, but I would prefer to check the error channel of the command itself on the select statement at the bottom, instead of having the "racy" form.